src/main/java/ru/spbstu/model/Variant.java

```diff
@@ -27,37 +28,59 @@
 @Table(name = "variants", uniqueConstraints = @UniqueConstraint(columnNames = {"chrom", "pos", "ref", "alt"}))
 @Entity
 public class Variant {
-  @Id
-  @GeneratedValue(strategy = GenerationType.IDENTITY)
-  private Long id;
-  @Column(name = "chrom", nullable = false)
-  private String chromosome;
-  @Column(name = "pos", nullable = false)
-  private Long position;
-  @Column(name = "ref")
-  private String referenceBase;
-  @Column(name = "alt")
-  private String alternateBase;
-  @Column(name = "variant_code", nullable = false)
-  private String variantCode;
-  @ToString.Exclude
-  @EqualsAndHashCode.Exclude
-  @OneToMany(mappedBy = "variant", cascade = CascadeType.ALL,
-      orphanRemoval = true)
-  private List<Annotation> annotations = new ArrayList<>();
-
-  public Variant addAnnotation(Annotation annotation) {
-    this.annotations.add(annotation);
-    annotation.setVariant(this);
-    return this;
-  }
-
-  // TODO: remove this after migrating to PostgreSQL
-  @PreUpdate
-  @PrePersist
-  private void updateVariantCode() {
-    variantCode = Stream.of(this.chromosome, ":", position.toString(), ":", referenceBase, ">", alternateBase)
-        .filter(Objects::nonNull)
-        .collect(Collectors.joining());
-  }
+    @Id
+    @EqualsAndHashCode.Exclude
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @Column(name = "chrom", nullable = false)
+    private String chromosome;
+
+    @Column(name = "pos", nullable = false)
+    private Long position;
+
+    @Column(name = "ref")
+    private String referenceBase;
+
+    @Column(name = "alt")
+    private String alternateBase;
+
+    @Column(name = "variant_code", nullable = false)
+    private String variantCode;
+
+    @ToString.Exclude
+    @EqualsAndHashCode.Exclude
+    @OneToMany(mappedBy = "variant", cascade = CascadeType.ALL, orphanRemoval = true)
+    private Set<Annotation> annotations = new HashSet<>();
+
+    public Variant addAnnotation(final Annotation annotation) {
+        annotation.setVariant(this);
+        annotations.add(annotation);
+        return this;
+    }
+
+    // TODO: remove this after migrating to PostgreSQL
+    @PreUpdate
+    @PrePersist
+    private void updateVariantCode() {
+        variantCode = Stream.of(this.chromosome, ":", position.toString(), ":", referenceBase, ">", alternateBase)
+                .filter(Objects::nonNull)
+                .collect(Collectors.joining());
+    }
+
+    public static Variant from(final VcfRecord vcfRecord,
```

Comment:
I think method signature doesn't fully reflect what this method does. I'd suggest to make it more understandable and differentiable from the below one.